muffnn/mlp/mlp_classifier.py

```diff
@@ -105,28 +105,35 @@ def _init_model_output(self, t):
 
         if self.multilabel_:
             self.input_targets_ = \
-                tf.placeholder(tf.int64, [None, self.n_classes_], "labels")
+                tf.placeholder(tf.float32, [None, self.n_classes_], "targets")
             self.output_layer_ = tf.nn.sigmoid(t)
+            self.zeros=tf.zeros_like(self.output_layer_)
+
         elif self.n_classes_ > 2:
-            self.input_targets_ = tf.placeholder(tf.int64, [None], "targets")
+            self.input_targets_ = tf.placeholder(tf.float32, [None], "targets")
             self.output_layer_ = tf.nn.softmax(t)
         else:
-            self.input_targets_ = tf.placeholder(tf.int64, [None], "targets")
+            self.input_targets_ = tf.placeholder(tf.float32, [None], "targets")
             t = tf.reshape(t, [-1])  # Convert to 1d tensor.
             self.output_layer_ = tf.nn.sigmoid(t)
         return t
 
     def _init_model_objective_fn(self, t):
         if self.multilabel_:
             cross_entropy = tf.nn.sigmoid_cross_entropy_with_logits(
-                t, tf.cast(self.input_targets_, np.float32))
+                t, tf.cast(self.input_targets_, tf.float32))
         elif self.n_classes_ > 2:
             cross_entropy = tf.nn.sparse_softmax_cross_entropy_with_logits(
                 t, self.input_targets_)
         else:
             cross_entropy = tf.nn.sigmoid_cross_entropy_with_logits(
                 t, tf.cast(self.input_targets_, np.float32))
-        self._obj_func = tf.reduce_mean(cross_entropy)
+        y_finite=tf.equal(self.input_targets_, -1)
+        self._selections=tf.select(y_finite, self.zeros, cross_entropy )
+        self._yfinite=y_finite
+        self.ce=cross_entropy
+        self._obj_func =  tf.reduce_sum(tf.select(y_finite,  self.zeros,cross_entropy ))
```

Comment:
We should build the zeros on the fly here, not in the training function above (which is actually part of the base class). You can use tf.zeros_like to get a Tensor of zeros that is the same shape as the input targets.

Remove the extra attributes we set during debugging, like `ce`, `selections`, `_yfinite`.